DfciPkg/SettingsManager/SettingsManagerProvider.c

```diff
@@ -359,21 +359,21 @@ ProviderValueAsAscii (
 
   switch (Provider->Type) {
     case DFCI_SETTING_TYPE_ENABLE:

```

Comment:
This design seems a bit dangerous.

Having the caller manually associate a C type and the buffer size (in most cases derived from the C type) based on these decoupled DFCI enum values increases likelihood of type mismatches like those in the file.

If the API truly provides the proper size for the provider's data, shouldn't all of these cases do what the `DFCI_SETTING_TYPE_STRING` case does and initially call with a zero sized buffer to get back the proper size to hold the data?

Reply:
I think improving this would require a larger API overhaul. Since you're just working on [#45](https://github.com/microsoft/mu_feature_dfci/issues/45), you can mark this resolved.

Reply:
I agree that the design needs a more thorough review.

In searching through the settings providers, I did not find any that were acting in an unsafe manner, but the interfaces still allow for too much ambiguity

Reply:
DFCI_SETTING_PROVIDER_GET functions check the ValueSize to ensure that it is large enough to fill the required data size, and if they do not, they supply back the EFI_BUFFER_TOO_SMALL and an updated ValueSize.

I believe this is to require the caller to handle memory allocation so that a setting provide code could be used in PEI instances.